File: src/main/java/uk/gov/ukho/battleshipsboot/model/ships/Submarine.java
package uk.gov.ukho.battleshipsboot.model.ships;

import uk.gov.ukho.battleshipsboot.model.game.BoardPosition;
import uk.gov.ukho.battleshipsboot.model.game.Orientation;

public class Submarine extends Ship {
    public Submarine(Orientation orient, BoardPosition boardPosition) {
        super(3, orient, boardPosition);
    }
}

File: src/main/java/com/harragan/battleshipsboot/model/kotlinmodel/ships/Submarine.kt
package com.harragan.battleshipsboot.model.kotlinmodel.ships

import com.harragan.battleshipsboot.model.kotlinmodel.game.BoardPosition
import com.harragan.battleshipsboot.model.kotlinmodel.game.Orientation

data class Submarine @JvmOverloads constructor (
        override val orientation : Orientation,
        override val boardPosition: BoardPosition,
        override val occupiedBoardPositions : List<BoardPosition>,
        override val length : Int = 3,
        override val id: Int,
        override val isSunk: Boolean = false
) : Ship